features/README.md

```diff
@@ -0,0 +1,109 @@
+This is an experimental feature suite in [Gherkin](https://cucumber.io/docs/reference),
+to write [Specifications by Example](https://en.wikipedia.org/wiki/Specification_by_example).
+It's a work in progress, and there's a lot of work ahead. Write a feature, and dive in !
+
+``` gherkin
+Feature: Writing gherkin scenarios
+In order to describe the feature I want
+As a contributor
+I want to be able to write scenarios and steps
+
+Feature: Running gherkin scenarios
+In order to develop quickly and fearlessly
+As a developer
+I want to be able to run the feature suite
+```
+
+Ideally running the feature suite should not ruin the dev database, _(unless we want it to)_
+and it should be done using a meteor command such as `test-features`.
+
+
+## WARNING
+
+Running the feature suite will TRUNCATE THE DATABASE before each Scenario.
+You WILL LOSE EVERYTHING on your development environment database.
```

Comment:
I'm 100% for using a `test_` database, but I have not managed to do so yet. Keep in mind that I'm very new to Meteor and server-side javascript. I'm not too fond of javascript because it violates elementary rules like the transitivity of equality and the commutativity of addition, but that's a sore subject for another debate… If we can manage to configure the test suite to use its own database, let's do it !

Reply:
Great!, some new script in `package.json` like:
```json
    "start:test-chimp": "MONGO_URL=mongodb://localhost:27017/sovereigntestchimp meteor --settings=config/development/settings.json",
```
works great:
```
3 scenarios (3 passed)
29 steps (29 passed)
```
using `sovereigntestchimp` specific database, so we can trash it without use the default dev database instance.

Reply:
Yes, I prefer to add it to the PR, but up to you.

Anyway, this works for me:
```json
    "start:for-features": "MONGO_URL=mongodb://localhost:27017/sovereingtestchimp meteor --settings=config/development/settings.json --port 3003",
    "test:features": "./node_modules/.bin/chimp --chai --ddp=http://localhost:3003",
```
maybe your error it's because of your npm version? Here:
```bash
$ npm -v
5.6.0
$ meteor npm -v
5.6.0
```

I don't know how to do it in one line :-/ I usually run meteor in background in our CI, sleep 120, and later I run the tests that depends on it. Finally I stop meteor. Maybe I can improved it looking for the meter port in a loop:
```bash
# this inside a while til some timeout or 3003 start to listen
   lsof -i:3003 > /dev/null 2>&1
   # check if $? equal 0, so 3003 is listening, so we can start our tests
```

Reply:
Ok, so it's a problem on my end, thanks for the tip !

```
$ npm -v
3.5.2
$ meteor npm -v
5.6.0
```

---

I'm investigating, but in case it's an obvious problem and you have the solution handy, I'll just dump here what I've got :

``` bash
$ meteor npm run test:features

> Sovereign.app@ test:features /home/domi41/code/snd/sovereign
> chimp --chai --ddp=http://localhost:3003

(node:4986) [DEP0022] DeprecationWarning: os.tmpDir() is deprecated. Use os.tmpdir() instead.

Master Chimp and become a testing Ninja! Check out our course: http://bit.ly/2btQaFu


[chimp] Running...
## There is an issue with `node-fibers` ##
`/home/domi41/code/snd/sovereign/node_modules/cucumber/node_modules/fibers/bin/linux-x64-57/fibers.node` is missing.

Try running this to fix the issue: /home/domi41/.meteor/packages/meteor-tool/.1.6.1.89za4a.ppaex++os.linux.x86_64+web.browser+web.cordova/mt-os.linux.x86_64/dev_bundle/bin/node /home/domi41/code/snd/sovereign/node_modules/cucumber/node_modules/fibers/build
/home/domi41/code/snd/sovereign/node_modules/cucumber/node_modules/fibers/fibers.js:20
	throw new Error('Missing binary. See message above.');
	^

Error: Missing binary. See message above.
    at Object.<anonymous> (/home/domi41/code/snd/sovereign/node_modules/cucumber/node_modules/fibers/fibers.js:20:8)
    at Module._compile (module.js:643:30)
    at Object.Module._extensions..js (module.js:654:10)
    at Module.load (module.js:556:32)
    at tryModuleLoad (module.js:499:12)
    at Function.Module._load (module.js:491:3)
    at Module.require (module.js:587:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/home/domi41/code/snd/sovereign/node_modules/cucumber/lib/cucumber/util/run.js:1:75)
    at Module._compile (module.js:643:30)
Cucumber steps failed
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! Sovereign.app@ test:features: `chimp --chai --ddp=http://localhost:3003`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the Sovereign.app@ test:features script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/domi41/.npm/_logs/2018-04-09T16_45_47_914Z-debug.log


domi41@idao:~/code/snd/sovereign$ /home/domi41/.meteor/packages/meteor-tool/.1.6.1.89za4a.ppaex++os.linux.x86_64+web.browser+web.cordova/mt-os.linux.x86_64/dev_bundle/bin/node /home/domi41/code/snd/sovereign/node_modules/cucumber/node_modules/fibers/build
gyp info it worked if it ends with ok
gyp info using node-gyp@3.4.0
gyp info using node@6.11.4 | linux | x64
gyp info spawn /usr/bin/python2
gyp info spawn args [ '/usr/share/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/home/domi41/code/snd/sovereign/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/usr/share/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/usr/include/nodejs/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/usr/include/nodejs',
gyp info spawn args   '-Dnode_gyp_dir=/usr/share/node-gyp',
gyp info spawn args   '-Dnode_lib_file=node.lib',
gyp info spawn args   '-Dmodule_root_dir=/home/domi41/code/snd/sovereign',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp: binding.gyp not found (cwd: /home/domi41/code/snd/sovereign) while trying to load binding.gyp
gyp ERR! configure error 
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/usr/share/node-gyp/lib/configure.js:304:16)
gyp ERR! stack     at emitTwo (events.js:106:13)
gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:219:12)
gyp ERR! System Linux 4.13.0-38-generic
gyp ERR! command "/usr/bin/node" "/usr/bin/node-gyp" "rebuild" "--release"
gyp ERR! cwd /home/domi41/code/snd/sovereign
gyp ERR! node -v v6.11.4
gyp ERR! node-gyp -v v3.4.0
gyp ERR! not ok 
node-gyp exited with code: 1
Please make sure you are using a supported platform and node version. If you
would like to compile fibers on this machine please make sure you have setup your
build environment--
Windows + OS X instructions here: https://github.com/nodejs/node-gyp
Ubuntu users please run: `sudo apt-get install g++ build-essential`
Alpine users please run: `sudo apk add python make g++`

```

Note that running the command by itself works, so it's probably a version mismatch problem, as you suspected.

---

And finally, since speed is not an issue during CI, if a sleep works I'm not against using that until we cook up a more complex bash script. One thing at a time ; the first thing is to start using BDD. Then, we can do it right :angel: